p2p/pex/pex_reactor.go

```diff
@@ -372,6 +364,37 @@ func (r *PEXReactor) ensurePeers() {
 	}
 }
 
+func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) {
+	// 1s == (1e9 ns) == (1 Billion ns)
+	billionNs := float64(time.Second.Nanoseconds())
+
+	// exponential backoff if it's not our first attempt to dial given address
+	var attempts int
+	if lAttempts, attempted := r.attemptsToDial.Load(addr.DialString()); attempted {
+		attempts = lAttempts.(int)
+		jitterSeconds := time.Duration(rand.Float64() * billionNs)
+		backoffDuration := jitterSeconds + ((1 << uint(attempts)) * time.Second)
+		r.Logger.Debug(fmt.Sprintf("Dialing %v", addr), "attempts", attempts, "backoff_duration", backoffDuration)
+		time.Sleep(backoffDuration)
```

Comment:
Do we have logic that limits the number of attempts per address? 

If this backoffDuration gets big, then these go-routines will just hang around for a while. So maybe we should be using a select block here with a ticker so we can break out on Quit ?